File: OneDrive/Desktop/COEXIST/FrontCoexist-final1Angular/FrontCoexist-final1/src/app/entity/Product.ts
import {TypeProduit} from "./TypeProduit";
import {TypeStatus} from "./TypeStatus";

export class Product {

    idProduct:number;
    img:string;
    description:string;
    productName:string;
    price:number;
    stock:number

    status?:TypeStatus;
    type?:TypeProduit;
  Date: string;

}

File: OneDrive/Desktop/COEXIST/backcoexist-final1SpringBoot/src/main/java/tn/esprit/coexist/entity/Product.java
package tn.esprit.coexist.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.ToString;

import java.util.Date;
import java.util.List;

@Entity
@Data
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private  Integer productId;
    private String productName;
    private String description;
    private String image;
    private Integer rate;
    private int stock;
    private float price;

    @Enumerated(EnumType.STRING)
    public TypeProduit type;
    public boolean isfavourite=false;
    @Temporal(TemporalType.TIMESTAMP)
    private Date dateAdded;

    @OneToMany(mappedBy = "product")
    @ToString.Exclude
    @JsonIgnore

    private List<OrderItem> orderItems;



}
